conans/client/rest/rest_client_v1.py

```diff
@@ -34,18 +35,26 @@ def router(self):
         return ClientV1Router(self.remote_url.rstrip("/"), self._artifacts_properties,
                               self._matrix_params)
 
-    def _download_files(self, file_urls):
+    def _download_files(self, file_urls, snapshot_md5):
         """
         :param: file_urls is a dict with {filename: url}
 
         Its a generator, so it yields elements for memory performance
         """
         downloader = FileDownloader(self.requester, None, self.verify_ssl, self._config)
+        download_cache = self._config.download_cache
+        if download_cache:
+            assert snapshot_md5 is not None, "if download_cache is set, we need the file checksums"
+            downloader = CachedFileDownloader(download_cache, downloader)
```

Comment:
I have tried that, but then it was very abstracted and difficult to follow, to make sure we are not changing functionality (for example, the ``output`` is None, because those file transfers are not reflected as downloads, but just to get some meta-info, like the manifests, or conaninfo). I agree this should be abstracted, but lets do that change later, when this feature is a bit more stabilized.